ut/client_ut.cpp

```diff
@@ -913,6 +915,168 @@ TEST_P(ClientCase, Query_ID) {
+    // Based on https://github.com/ClickHouse/clickhouse-cpp/issues/43
+    std::cerr << "Connected to: " << client_->GetServerInfo() << std::endl;
+    std::cerr << "DROPPING TABLE" << std::endl;
+    client_->Execute("DROP TEMPORARY TABLE IF EXISTS multiarray");
+
+    std::cerr << "CREATING TABLE" << std::endl;
+    client_->Execute(Query(R"sql(CREATE TEMPORARY TABLE IF NOT EXISTS multiarray
+    (
+        `arr` Array(Array(UInt64))
+    );
+)sql"));
+
+    std::cerr << "INSERTING VALUES" << std::endl;
+    client_->Execute(Query(R"sql(INSERT INTO multiarray VALUES ([[0,1,2,3,4,5], [100, 200], [10,20, 50, 70]]), ([[456, 789], [1011, 1213], [], [14]]), ([[]]);)sql"));
+    std::cerr << "INSERTED" << std::endl;
+
+    auto result = std::make_shared<ColumnArray>(std::make_shared<ColumnArray>(std::make_shared<ColumnUInt64>()));
+    ASSERT_EQ(0u, result->Size());
+
+    std::cerr << "SELECTING VALUES" << std::endl;
+    client_->Select("SELECT arr FROM multiarray", [&result](const Block& block) {
+        std::cerr << "GOT BLOCK: " << block.GetRowCount() << std::endl;
+        if (block.GetRowCount() == 0)
+            return;
+
+        result->Append(block[0]);
+    });
+
+    std::cerr << "DONE SELECTING VALUES" << std::endl;
+    client_.reset();
+
+    ASSERT_EQ(3u, result->Size());
+    {
+        // ([[0,1,2,3,4,5], [100, 200], [10,20, 50, 70]])
+        const std::vector<std::vector<uint64_t>> expected_vals = {
+            {0, 1, 2, 3, 4, 5},
+            {100, 200},
+            {10, 20, 50, 70}
+        };
+
+        auto row = result->GetAsColumnTyped<ColumnArray>(0);
+        ASSERT_EQ(3u, row->Size());
+        EXPECT_TRUE(CompareRecursive(expected_vals[0], *row->GetAsColumnTyped<ColumnUInt64>(0)));
+        EXPECT_TRUE(CompareRecursive(expected_vals[1], *row->GetAsColumnTyped<ColumnUInt64>(1)));
+        EXPECT_TRUE(CompareRecursive(expected_vals[2], *row->GetAsColumnTyped<ColumnUInt64>(2)));
+    }
+
+    {
+        // ([[456, 789], [1011, 1213], [], [14]])
+        const std::vector<std::vector<uint64_t>> expected_vals = {
+            {456, 789},
+            {1011, 1213},
+            {},
+            {14}
+        };
+
+        auto row = result->GetAsColumnTyped<ColumnArray>(1);
+        ASSERT_EQ(4u, row->Size());
+        EXPECT_TRUE(CompareRecursive(expected_vals[0], *row->GetAsColumnTyped<ColumnUInt64>(0)));
+        EXPECT_TRUE(CompareRecursive(expected_vals[1], *row->GetAsColumnTyped<ColumnUInt64>(1)));
+        EXPECT_TRUE(CompareRecursive(expected_vals[2], *row->GetAsColumnTyped<ColumnUInt64>(2)));
+        EXPECT_TRUE(CompareRecursive(expected_vals[3], *row->GetAsColumnTyped<ColumnUInt64>(3)));
+    }
+
+    {
+        // ([[]])
+        auto row = result->GetAsColumnTyped<ColumnArray>(2);
+        ASSERT_EQ(1u, row->Size());
+        EXPECT_TRUE(CompareRecursive(std::vector<uint64_t>{}, *row->GetAsColumnTyped<ColumnUInt64>(0)));
+    }
+}
+
+ColumnRef RoundtripColumnValues(Client& client, ColumnRef expected) {
```

Comment:
What do you think about renaming this to something like `InsertAndQuery`? I understand the reasoning behind Roundtrip, but it doesn't seem to be the most intuitive one.